sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala

```diff
@@ -791,4 +791,28 @@ class AnalysisErrorSuite extends AnalysisTest {
       assertAnalysisError(plan, s"Correlated column is not allowed in predicate ($msg)" :: Nil)
     }
   }
+
+  test("SPARK-35618: Resolve star expressions in subquery") {
```

Comment:
I read the PR description and I thought this PR is to accept new query patterns, but this PR only has the negative test cases?

Reply:
Yes, currently only `Filter` can host outer references for correlated subqueries, and star expansion only happens when the node is either a `Project` or `Aggregate` (buildExpandedProjectList). It will be clearer with lateral subquery examples: 
```sql
// t: [a, b]
SELECT * FROM t, LATERAL (SELECT t.*)  // <--- t.* should be resolved as t.a, t.b
```